projects/ngrx.io/content/guide/signals/signal-store/testing.md

```diff
@@ -0,0 +1,687 @@
+
+```typescript
+describe('withTrackedPlay', () => {
+  const TrackedPlayStore = signalStore({providedIn: 'root'}, withPlayTracking();
+
+  afterEach(async () => {
+    await jest.runOnlyPendingTimersAsync()
+    jest.useRealTimers();
+  })
+
+  it('should track movies', () => {
+    useFakeTimers()
+    const store = TestBed.inject(TrackedPlayStore)
+
+    store.play(1);
+    jest.advanceTimersByTime(1000);
+
+    store.stop();
+    store.play(2)
+    jest.advanceTimersByTime(1000);
+
+    store.play(3);
+    jest.advanceTimersByTime(1000);
+
+    store.play(1)
+    jest.advanceTimersByTime(1000);
+    store.stop()
+
+    expect(store.trackedData()).toEqual({1: 2000, 2: 1000, 3: 1000});
+  })
+});
+```
+
+## Mocking the Signal Store
+
+What applies to testing the Signal Store itself also applies to mocking it. The Signal Store behaves like any other service, meaning it can be mocked using the same tools and techniques used for other services.
+
+A `MovieComponent` uses the `MoviesStore` to display the movies:
+
+```typescript
+@Component({
+  selector: 'app-movies',
+  template: ` <input type="text" [(ngModel)]="studio" [disabled]="store.loading()" placeholder="Name of Studio" />
+    <ul>
+      @for (movie of store.movies(); track movie.id) {
+      <p>{{ movie.id }}: {{ movie.name }}</p>
+      }
+    </ul>`,
+  standalone: true,
+  imports: [FormsModule],
+})
+export class MoviesComponent {
+  protected studio = signal('');
+  protected readonly store = inject(MoviesStore);
+
+  constructor() {
+    this.store.load(this.studio);
+  }
+}
+```
+
+### Native Mocking
+
+```typescript
+it('should show movies (native Jest)', () => {
+  const load = jest.fn<void, [Signal<string>]>();
+
+  const moviesStore = {
+    movies: signal(new Array<Movie>()),
+    loading: signal(false),
+    load,
+  };
+
+  TestBed.configureTestingModule({
+    imports: [MoviesComponent],
+    providers: [
+      {
+        provide: MoviesStore,
+        useValue: moviesStore,
+      },
+    ],
+  });
+
+  const fixture = TestBed.createComponent(MoviesComponent);
+  fixture.autoDetectChanges(true);
+
+  const studio = load.mock.calls[0][0];
+  const input: HTMLInputElement = fixture.debugElement.query(By.css('input')).nativeElement;
+
+  expect(studio()).toBe('');
+
+  input.value = 'Warner Bros';
+  input.dispatchEvent(new Event('input'));
+  expect(studio()).toBe('Warner Bros');
+
+  moviesStore.movies.set([
+    { id: 1, name: 'Harry Potter' },
+    { id: 2, name: 'The Dark Knight' },
+  ]);
+  fixture.detectChanges();
+
+  const movieNames = fixture.debugElement.queryAll(By.css('p')).map((el) => el.nativeElement.textContent);
+  expect(movieNames).toEqual(['1: Harry Potter', '2: The Dark Knight']);
+});
+```
+
+The test mocks only the properties and methods that are used by the component in that particular test. Even if a Signal Store has additional methods, it is not necessary to mock all of them.
+
+### ng-mocks
```

Comment:
I make the assumption that this is added because of the popularity of ng-mocks.
But, I don't think this adds much value here and can be removed.
We already provide good examples and have a mention/link to `ng-mocks`.